examples/5_secret_storage/secret_storage.go

```diff
@@ -0,0 +1,535 @@
+	if !ok || err != nil {
+		panic(fmt.Sprintf("Unexpected result: %v, %v", ok, err))
+	}
+
+	res, err = rental.Lookup(team, tool)
+	fmt.Printf("LOOKUP: %+v, %v\n", res, err)
+
+	ok, res, err = rental.Unreserve(team, user1, tool, date3)
+	fmt.Printf("UNRESERVE: %v, %+v, %v\n", ok, res, err)
+	if !ok || err != nil {
+		panic(fmt.Sprintf("Unexpected result: %v, %v", ok, err))
+	}
+
+	ok, res, err = rental.Unreserve(team, user1, tool, date2)
+	fmt.Printf("EXPECTING UNRESERVE FAIL: %v, %+v, %v\n", ok, res, err)
+	if ok && err == nil {
+		panic(fmt.Sprintf("Unexpected result: %v, %v", ok, err))
+	}
+
+	ok, res, err = rental.Unreserve(team, user1, tool, date1)
+	fmt.Printf("UNRESERVE: %v, %+v, %v\n", ok, res, err)
+	if !ok || err != nil {
+		panic(fmt.Sprintf("Unexpected result: %v, %v", ok, err))
+	}
+
+	res, err = rental.Lookup(team, tool)
+	fmt.Printf("LOOKUP: %+v, %v\n", res, err)
+}
+
+func concurrentRentalUser(rental RentalBotClient, team string, tool string, userID int, wg *sync.WaitGroup) {
+	date := fmt.Sprintf("2044-10-0%d", userID)
+	user := fmt.Sprintf("user%d", userID)
+
+	i := 0
+	// keep trying to reserve for user's unique date until successful
+	for {
+		ok, res, err := rental.Reserve(team, user, tool, date)
+		i++
+		fmt.Printf("%v, attempt %d, TRY TO RESERVE: %v, %+v, %+v\n", user, i, ok, res, err)
+		if ok && err == nil {
+			break
+		}
+	}
+	wg.Done()
+}
+
+func concurrentRentalUsers(rental RentalBotClient, team string) {
+	tool := "time travel machine"
+
+	var wg sync.WaitGroup
+
+	// pre
+	wg.Add(1)
+	go func() {
+		for {
+			ok, res, err := rental.Remove(team, tool)
+			fmt.Printf("TRY TO REMOVE: %v, %+v, %+v\n", ok, res, err)
+			if ok && err == nil {
+				break
+			}
+		}
+		wg.Done()
+	}()
+	wg.Wait()
+
+	// have 5 users concurrently try to reserve the same tool for 5 unique dates
+	for i := 1; i <= 5; i++ {
+		wg.Add(1)
+		go concurrentRentalUser(rental, team, tool, i, &wg)
+	}
+	wg.Wait()
+
+	// post: check that the tool has been reserved for all 5 unique dates
+	var val map[string]string
+	res, err := rental.Lookup(team, tool)
+	if err != nil {
+		panic(err)
+	} else if res.EntryValue == "" {
+		val = map[string]string{}
+	} else {
+		if err := json.Unmarshal([]byte(res.EntryValue), &val); err != nil {
+			panic(err)
+		}
+	}
+	if len(val) != 6 {
+		panic(fmt.Sprintf("Unexpected result: %+v\n", val))
+	}
+
+}
+
+func main() {
+	const MsgPrefix = "!storage"
+
+	var kbLoc string
+	var kbc *kbchat.API
+	var err error
+
+	flag.StringVar(&kbLoc, "keybase", "keybase", "the location of the Keybase app")
+	flag.Parse()
+
+	fmt.Println("Starting 5_secret_storage example...")
+
+	if kbc, err = kbchat.Start(kbchat.RunOptions{KeybaseLocation: kbLoc}); err != nil {
+		fail("Error creating API: %s", err.Error())
+	}
+
+	team := "yourhackerspace"
+
+	secretClient := new(SecretKeyKVStoreAPI)
+	secretClient.api = kbc
+	rental := RentalBotClient{api: secretClient}
+
+	fmt.Println("...basic rental actions...")
+	basicRentalUsers(rental, team)
```

Comment:
If these calls return an error you can `log.Fatalf` instead of having panics throughout the code